### DISCUSSION TITLE
How to document a script setup component with JSDoc?

### DISCUSSION THREAD

**Discussion by rodrigocfd** [vuejs/language-tools#2383] [Q&A]
I can use JSDoc to document a function:

```ts
/**
 * This function does something.
 */
export function doSomething() { }
```

...and when I hover this function in VSCode, it shows the description nicely.

Now suppose I have a simple `script setup` component:

```vue
<script setup lang="ts">
</script>
```

How can I document this component with JSDoc, so that Volar displays the description in VSCode, just like the `doSomething` function above?

**Comment by xiaoxiangmoe** [comment]
```vue
<script lang="ts">
/**
 * # My Button Component
 * To trigger an operation.
 * 
 * @deprecated Please use `MyButtonV2` instead.
 * 
 * @example
 * ```vue
 * <MyButton>Click Me</MyButton>
 * ```
 */
export default {};
</script>
<script setup lang="ts">
defineProps<{
  /**
   * Set button type
   */
  buttonType?: 'primary' | 'dashed' | 'link' | 'text' | 'default';
}>();
</script>
<template>
  <button></button>
</template>
```
<img width="702" alt="image" src="https://github.com/vuejs/language-tools/assets/8111351/315eebbc-9d4b-461d-bf73-b152d1d123b9">
<img width="441" alt="image" src="https://github.com/vuejs/language-tools/assets/8111351/e5f41bf8-70e1-4f9f-a415-22ac34f2a01a">
<img width="615" alt="image" src="https://github.com/vuejs/language-tools/assets/8111351/f48b451d-b339-4e16-899b-d6d4679d786f">

**Reply by johnsoncodehk** [reply]
For JSDoc support in template, please track for #3651.
